.claude/hooks/enforce-uv.sh

```
@@ -0,0 +1,141 @@
+      
+      case "$pip_cmd" in
+        install\ *)
+          packages=$(echo "$pip_cmd" | sed 's/install//' | sed 's/--[^ ]*//g' | xargs)
+          
+          # -r requirements.txt
+          if [[ "$pip_cmd" =~ -r\ .*\.txt ]]; then
+            req_file=$(echo "$pip_cmd" | sed -n 's/.*-r \([^ ]*\).*/\1/p')
+            echo "{
+              \"decision\": \"block\",
+              \"reason\": \"📋 requirements.txtからインストール:\\n\\n✅ 推奨方法:\\nuv add -r $req_file\\n\\nこれにより:\\n• requirements.txt内のすべての依存関係をpyproject.tomlに追加\\n• uv.lockファイルを自動生成/更新\\n• 仮想環境を自動的に同期\\n\\n💡 制約ファイルがある場合:\\nuv add -r $req_file -c constraints.txt\\n\\n📌 注意: この方法が最も確実で、バージョン指定も正しく処理されます\"
+            }"
```

Shellcheck (0.10.0)

[info] echo may not expand escape sequences. Use printf.

(SC2028)